### PR TITLE
Check if TAG exists before IMAGE_TAG is assigned.

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind failing-test

**What this PR does / why we need it**:

Check if TAG exists before IMAGE_TAG is assigned.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
None.
```

### CODE REVIEW

**Comment by feiskyer** [comment]
There're still some configuration errors:

```
W1231 08:19:53.343] 2019/12/31 08:19:53 process.go:155: Step 'bash -c . hack/lib/version.sh && KUBE_ROOT=. kube::version::get_version_vars && echo "${KUBE_GIT_VERSION-}"' finished in 1.548097335s
W1231 08:19:53.345] 2019/12/31 08:19:53 main.go:319: Something went wrong: failed to acquire k8s binaries: -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.0.0-20190123011202-457680cc0804/azblob/zc_storage_error.go:42
W1231 08:19:53.345] ===== RESPONSE ERROR (ServiceCode=ContainerNotFound) =====
W1231 08:19:53.345] Description=The specified container does not exist.
```

File: Makefile [hunk 1]
```diff
@@ -40,7 +40,13 @@ IMAGE_REGISTRY ?= local
 STAGING_REGISTRY := gcr.io/k8s-staging-provider-azure
 K8S_VERSION ?= v1.15.0
 HYPERKUBE_IMAGE ?= gcrio.azureedge.net/google_containers/hyperkube-amd64:$(K8S_VERSION)
-IMAGE_TAG ?= $(shell git rev-parse --short=7 HEAD)
+
+ifndef TAG
+	IMAGE_TAG = $(shell git rev-parse --short=7 HEAD)
```
Comment (chewong) [inline Makefile]:
```suggestion
	IMAGE_TAG ?= $(shell git rev-parse --short=7 HEAD) 
```
Using a conditional variable assignment operator is necessary to pick up environment variables set in https://github.com/kubernetes/test-infra/blob/master/kubetest/azure.go#L885-L887, which are used to build customm ccm and cnm according to the PR and run e2e test with them.

Comment (nilo19) [reply Makefile]: Thank you @chewong .

**Pull request merged**